embedded/sql/engine.go

```diff
@@ -1116,9 +1125,32 @@ func (e *Engine) ExecPreparedStmts(stmts []SQLStmt, params map[string]interface{
 			return nil, nil, ErrIllegalArguments
 		}
 
+		_, isDBSelectionStmt := stmt.(*UseDatabaseStmt)
+
+		// handle the case when working in non-autocommit mode outside a transaction block
+		if isDBSelectionStmt && (currTx != nil && !currTx.closed) && !currTx.explicitClose {
+			err = currTx.commit()
```

Comment:
I'm still not sure if this is a good idea,

IIRC this would only happen in such situation:

* autocommit is set to false
* there are some statements already executed - thus this is like an implicit transaction
* some use db statement is given

What worries me here is that this is a place where implicit commit happens. What if use database or any of the statements that follow fail? That would mean that we've executed part of the statements and failed with others.

Reply:
when autocommit is disabled, the exact number of transactions may be unknown upfront. In case there is an ongoing tx and one of the following statements if next: `BEGIN TRANSACTION`,  `USE DATABASE`, the ongoing tx is committed and a new one is initialized